### PR TITLE
Refactor/config

### PR DESCRIPTION
**Pull request by loresuso**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind flaky-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area library

/area cli

> /area tests

> /area examples

**What this PR does / why we need it**:

This PR make naming more smart: the basic idea is to map the config keys to the commands. 
A new possible configuration will now be for instance:

```
registry:
  auth:
    basic: []
    oauth: []

artifact:
  follow:
    refs:
    - falco-rules:1
    - application-rules:1
    every: 24h0m0s
    falcoversions: http://localhost:8765/versions
    plugins-dir:
  install:
    refs: [pippo]
    pluginsdir: /pluto
    rulesfiles-dir:

indexes:
- name: falcosecurity
  url: https://falcosecurity.github.io/falcoctl/index.yaml
```
As you can see, the structure of the yaml file resembles the commands. Env variables will move accordingly.

TODO: Move login and oauth to `auth login basic` and `auth login oauth`. Also, create a shortcut from `falcoctl login` to `auth login basic`. 

Commits are kept as much separated as possible during the refactor so we can better keep track of regressions. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: internal/registry/auth/basic/basic.go [hunk 1]
```diff
@@ -43,14 +43,14 @@ func (o *loginOptions) Validate(args []string) error {
 	return nil
 }
 
-// NewLoginCmd returns the login command.
+// NewLoginCmd returns the basic command.
 func NewLoginCmd(ctx context.Context, opt *options.CommonOptions) *cobra.Command {
```
Comment (leogr) [inline internal/registry/auth/basic/basic.go:47]:
```suggestion
func NewBasicCmd(ctx context.Context, opt *options.CommonOptions) *cobra.Command {
```

File: internal/registry/auth/basic/basic.go [hunk 2]
```diff
@@ -43,14 +43,14 @@ func (o *loginOptions) Validate(args []string) error {
...
 	o := loginOptions{
 		CommonOptions: opt,
 	}
 
 	cmd := &cobra.Command{
-		Use:                   "login hostname",
+		Use:                   "basic hostname",
```
Comment (leogr) [inline internal/registry/auth/basic/basic.go:53]:
```suggestion
		Use:                   "basic <hostname>",
```

**Review by leogr — COMMENTED** [review]
Overall SGTM

Just found a func name that has not yet been renamed.

**Comment by loresuso** [comment]
Thank you @leogr! Comments addressed!

**Review by leogr — APPROVED** [review]
:+1:

**Pull request merged**